### ISSUE TITLE
warn on invocations of `globals()`

### ISSUE THREAD

**Issue by kiaradlf** [astral-sh/ruff#12845]
python has a number of introspection functions, among which [`globals()`](https://docs.python.org/3/library/functions.html#globals) and [`locals()`](https://docs.python.org/3/library/functions.html#locals).

these can currently be used both for reading of variables as well as for assignment purposes, say:

```python
for foo in foos:
  globals()[foo] = ...
```

while it seems that ruff currently lets these slide, i believe that for both legibility as well as to facilitate static analysis, use of such invocations is considered an anti-pattern.
it would be nice to see ruff suggest users to reconsider using the likes of these.

**Comment by Avasam** [comment]
Relates to https://github.com/astral-sh/ruff/issues/2031#issuecomment-2294987091 as well

**Comment by MichaReiser** [comment]
That seems a reasonable "restriction" rule to me. Another relevant rule for `locals` seems to lint against mutations on `assignments` (although that's also something a type checker is good at catching).

I'm leaning towards deferring the implementation of such a rule until #1774 is done because there are valid use cases for both expressions. (@AlexWaygood what's your take?)
